### ISSUE TITLE
Repeated if merge breaks code when an expression variable changes after the first if block

### ISSUE THREAD

**Issue by JeremyStepp** [sourcery-ai/sourcery#147]
### Issue description or question
the `Merge repeated if statements into single if` refactor can change functionality if the code run in the first if block changes the variable that is being checked.

Here is an example:
```
# If game settings haven't been generated
if not game_settings.current_game:
    start_setup_wizard()

# Close program if the user left the setup wizard
# without generating the game settings
if not game_settings.current_game:
    sys.exit()
```

### Sourcery Version
0.93

### Code editor or IDE name and version
VSCode 1.58.2

**Comment by Hellebore** [comment]
Thanks for raising - can you give a little more context on how game_settings is scoped in your example?

**Comment by JeremyStepp** [comment]
It's an object imported from the package __init__.py file. A setup_wizard module seperate from the code shown ends up editing it by rerunning the object's __init__ function after making changes to the config file it loads from.

**Comment by Hellebore** [comment]
Fixed in 0.9.5
